pkg/pod/container.go

```diff
@@ -280,6 +280,25 @@ func (builder *ContainerBuilder) WithResourceRequest(hugePages, memory string, c
 	return builder
 }
 
+// WithCustomResourcesRequests applies custom resource requests struct on container.
+func (builder *ContainerBuilder) WithCustomResourcesRequests(resourceList corev1.ResourceList) *ContainerBuilder {
+	glog.V(100).Infof("Applying custom resource requests to container: %v", resourceList)
+
+	if len(resourceList) == 0 {
+		glog.V(100).Infof("Container's resource limit var 'resourceList' is empty")
+
+		builder.errorMsg = "container's resource limit var 'resourceList' is empty"
+	}
+
+	if builder.errorMsg != "" {
```

Comment:
I think this can be updated based on https://github.com/openshift-kni/eco-goinfra/pull/355/files 

If there is already an error, we can return directly the builder in the previous if block.

Reply:
thanks, updated